presentation-api/receiving-ua/support/PresentationConnection_onclose_receiving-ua.html

```diff
@@ -0,0 +1,79 @@
+    assert_true(evt instanceof PresentationConnectionCloseEvent, 'An event using PresentationConnectionCloseEvent is fired.');
+    assert_true(evt.isTrusted, 'The event is a trusted event.');
+    assert_false(evt.bubbles, 'The event does not bubbles.');
+    assert_false(evt.cancelable, 'The event is not cancelable.');
+    assert_equals(evt.type, 'close', 'The event name is "close".');
+    assert_equals(evt.target, connection, 'event.target is the presentation connection.');
+    assert_equals(connection.state, 'closed', 'State of the presentation connection is "closed".');
+    assert_equals(evt.reason, reason, 'The reason for closing the presentation connection is "' + reason + '".');
+  };
+
+  promise_test(t => {
+    return navigator.presentation.receiver.connectionList.then(list => {
+      let connection = list.connections[0];
+      assert_equals(list.connections.length, 1, 'A presentation connection list is populated with a first presentation connection.');
+
+      // Step 1: close the presentation connection in "connected" state
+      connection.close();
+      let watchClose = new EventWatcher(t, connection, 'close');
+      const watchNewConnection = new EventWatcher(t, list, 'connectionavailable');
+      return watchClose.wait_for('close').then(evt => {
+        checkCloseEvent(evt, connection, 'closed');
+
+        // Step 2: check a connection closed by the controlling user agent
+        stash.send(JSON.stringify({ type: 'ok' }));
+        return watchNewConnection.wait_for('connectionavailable')
+      }).then(evt => {
+        connection = evt.connection;
+        watchClose = new EventWatcher(t, connection, 'close');
+        return watchClose.wait_for('close');
+      }).then(evt => {
+        checkCloseEvent(evt, connection, 'closed');
+
+        // Step 3: try to close the presentation connection in "closed" state (nothing should happen)
+        connection.close();
+        return Promise.race([
+          new Promise(resolve => { t.step_timeout(resolve, 1000); }),
+          watchClose.wait_for('close').then(() => {
+            assert_unreached('Invoking PresentationConnection.close() in the "closed" state causes nothing.'); })
+        ]);
+      }).then(() => {
+
+        // Step 4: check a connection closed due to aborting the nested browsing context
+        stash.send(JSON.stringify({ type: 'ok' }));
+        return watchNewConnection.wait_for('connectionavailable');
+      }).then(evt => {
+        connection = evt.connection;
+        stash.send(JSON.stringify({ type: 'ok' }));
+        watchClose = new EventWatcher(t, connection, 'close');
+        return Promise.race([
+          watchClose.wait_for('close'),
+          new Promise(resolve => { t.step_timeout(() => { resolve(null); }, 3000); })
+        ]);
+      }).then(evt => {
+        assert_true(!!evt, 'A presentation connection is closed when its controller\'s browsing context is discarded.');
```

Comment:
Now that this SHOULD has been changed to MUST in the current spec, I'll cancel the change to "[Optional]".